build.gradle

```diff
@@ -129,6 +129,8 @@ dependencies {
     // https://mvnrepository.com/artifact/org.assertj/assertj-core
     testImplementation group: 'org.assertj', name: 'assertj-core', version: '3.17.2'
 
+    testImplementation group: 'org.springframework', name: 'spring-test'
```

Comment:
do we really need this? we are not using spring context in the code. I think this will make the project heavier

Reply:
For mocking multipart file, I think mockito should help by mocking the interface and it will dynamically create the implementation and that way you can return the values by using Mockito.when ... then construct.

Reply:
Good point. I introduced spring-test when I was trying to make the integration test work. When I moved on from that, I didn't think of simply going back to mockito.